src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +50,36 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo)
+            throws IllegalDateParametersException {
+
+        final DateTimeFormatter formatter =
+                DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        StringBuilder report = new StringBuilder("Отчёт за период ");
```

Comment:
it's better to create report after you validate data, cause there is no need to create it if date check doesn't pass